### PR TITLE
[flag_dice_1]: list reading and test

### PR DESCRIPTION
**Pull request by wsh-wood**
### What?

- brief: list reading and adding test.
- author: wsh-wood non-knife
### What changes are included in this PR?

Contains  doc folder and  src folder, doc folder is the learning task notes, src folder is the test file written.

### Are these changes tested?

image

task_1_win_2

### CODE REVIEW

File: stl/flag_dice/non_knife/doc/task_1/study_list.md [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+    const_iterator __i2 = __y.begin();
+    while (__i1 != __end1 && __i2 != __end2 && *__i1 == *__i2)
+{
+    ++__i1;
+    ++__i2;
+}
+    return __i1 == __end1 && __i2 == __end2;
+}
+```
+>实现思路是，迭代判断两个迭代器是否都抵达末尾。
```
Comment (Light-City) [inline stl/flag_dice/non_knife/doc/task_1/study_list.md:64]:
```
> blank your content.
```

File: stl/flag_dice/non_knife/src/task_1/main.cpp [hunk 2]
```diff
@@ -0,0 +1,131 @@
+#include <iostream>
+#include <list>
+#include <string>
+using namespace std;
```
Comment (Light-City) [inline stl/flag_dice/non_knife/src/task_1/main.cpp:4]: Does not comply with code specification

File: stl/flag_dice/non_knife/src/task_1/main.cpp [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+
+bool comparePerson(Person& p1, Person& p2) {
+  if (p1.m_Age == p2.m_Age) {
+    return p1.m_Height > p2.m_Height;
+  }
+  return p1.m_Age < p2.m_Age;
+}
+
+void test() {
+  list<Person> L;
```
Comment (Light-City) [inline stl/flag_dice/non_knife/src/task_1/main.cpp:100]: Does not comply with code specification

File: stl/flag_dice/non_knife/src/task_1/main.cpp [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+  for (list<Person>::const_iterator it = L.begin(); it != L.end(); it++) {
+    cout << "姓名：" << it->m_Name << " 年龄：" << it->m_Age << " 身高：" << it->m_Height
+         << endl;
+  }
+}
+
+int main() {
+  test();
+
+  system("pause");
```
Comment (Light-City) [inline stl/flag_dice/non_knife/src/task_1/main.cpp:129]: remove this in linux.

File: stl/flag_dice/wsh_wood/doc/task_1/list.md [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+BM_PrefixDecrement/64    948 ns          948 ns       738462
+BM_PrefixDecrement/512  7.59 us         7.59 us        92160
+BM_PrefixDecrement/4k   60.6 us         60.6 us        11538
+BM_PrefixDecrement/8k    121 us          121 us         5769
+BM_PostfixDecrement/8    118 ns          118 ns      5933333
+BM_PostfixDecrement/64   949 ns          949 ns       738462
+BM_PostfixDecrement/512 7.60 us         7.60 us        92160
+BM_PostfixDecrement/4k   61.0 us         61.0 us       11538
+BM_PostfixDecrement/8k    122 us          122 us        5769
+```
```
Comment (Light-City) [inline stl/flag_dice/wsh_wood/doc/task_1/list.md:194]: good test.

File: stl/flag_dice/wsh_wood/doc/task_1/test.jpg
Comment (Light-City) [inline stl/flag_dice/wsh_wood/doc/task_1/test.jpg:1]: change test.jpg to a spec name.

File: stl/flag_dice/wsh_wood/doc/task_1/list.md [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+```C++
+Run on (4 X 2400 MHz CPU s)
+CPU Caches:
+  L1 Data 32 KiB (x2)
+  L1 Instruction 32 KiB (x2)
+  L2 Unified 256 KiB (x2)
+  L3 Unified 3072 KiB (x1)
+Load Average: 0.25, 0.18, 0.12
+------------------------------------------------------------
+Benchmark                  Time             CPU   Iterations
```
Comment (Light-City) [inline stl/flag_dice/wsh_wood/doc/task_1/list.md:182]: Why didn't I see this piece of code?

Comment (wsh-wood) [reply stl/flag_dice/wsh_wood/doc/task_1/list.md:182]: I was surprised to find that there was no difference between the two types of '++'. So I contacted Bing AI again to confirm the response I received, so it was included in the reference section.

Comment (wsh-wood) [reply stl/flag_dice/wsh_wood/doc/task_1/list.md:182]: > Why didn't I see this piece of code?

I encountered some unresolved issues while compiling Google benchmark yesterday, so this section was not validated. I can supplement this section later.

File: stl/flag_dice/non_knife/doc/task_1/study_list.md (hunk 1 above)
Comment (wsh-wood) [reply stl/flag_dice/non_knife/doc/task_1/study_list.md]: The file has been removed.

File: stl/flag_dice/non_knife/src/task_1/main.cpp (hunk 4 above)
Comment (wsh-wood) [reply stl/flag_dice/non_knife/src/task_1/main.cpp]: Corrected.

File: stl/flag_dice/non_knife/src/task_1/main.cpp (hunk 3 above)
Comment (wsh-wood) [reply stl/flag_dice/non_knife/src/task_1/main.cpp]: Corrected

File: stl/flag_dice/wsh_wood/doc/task_1/test.jpg
Comment (wsh-wood) [reply stl/flag_dice/wsh_wood/doc/task_1/test.jpg:1]: Corrected.

File: stl/flag_dice/non_knife/src/task_1/main.cpp (hunk 2 above)
Comment (wsh-wood) [reply stl/flag_dice/non_knife/src/task_1/main.cpp]: Corrected.

File: stl/flag_dice/wsh_wood/doc/task_1/list.md (hunk 6 above)
Comment (wsh-wood) [reply stl/flag_dice/wsh_wood/doc/task_1/list.md]: The test code has been added and the MD file has been updated.

**Comment by Light-City** [comment]
flag_dice_3-> flag_dice_1

**Comment by wsh-wood** [comment]
> flag_dice_3-> flag_dice_1

ok

**Comment by wsh-wood** [comment]
> flag_dice_3-> flag_dice_1

ok

**Review by Light-City — COMMENTED** [review]
lgtm

**Pull request merged**